File: Microservices/ContactServices/src/models/contact.model.js
'use strict'

const mongoose = require('mongoose');

const Schema = mongoose.Schema;

const DOCUMENT_NAME = 'Contacts'

const HistorySchema = new Schema({
    createdBy: { type: String, required: true }, //id
    createdName: { type: String, required: true },
    description: { type: String, required: true },
    createdAt: { type: Date, default: Date.now },
})

const ContactSchema = new Schema({
    username: { type: String, required: true },
    email: { type: String, required: true },
    phone: { type: String, required: true },
    content: { type: String, required: true },
    isCheck: { type: Boolean, default: false },
    StatusActive: { type: Boolean, default: true },
    creator: [HistorySchema],
}, { minimize: false, timestamps: true })

const contactModel = mongoose.models.contact || mongoose.model(DOCUMENT_NAME, ContactSchema)

module.exports = contactModel;

File: MobileApp/siegenx_mobile_app/lib/models/contact_model.dart
class Contact {
  final int id;
  final String username;
  final String email;
  final String phone;
  final String content;
  final bool isCheck;
  final DateTime date;
  final DateTime createdAt;
  final DateTime updatedAt;

  Contact({
    required this.id,
    required this.username,
    required this.email,
    required this.phone,
    required this.content,
    required this.isCheck,
    required this.date,
    required this.createdAt,
    required this.updatedAt,
  });

  // Chuyển đổi từ JSON sang Object
  factory Contact.fromJson(Map<String, dynamic> json) {
    return Contact(
      id: json['id'],
      username: json['username'],
      email: json['email'],
      phone: json['phone'],
      content: json['content'],
      isCheck: json['isCheck'],
      date: DateTime.parse(json['date']),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  // Chuyển đổi từ Object sang JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'username': username,
      'email': email,
      'phone': phone,
      'content': content,
      'isCheck': isCheck,
      'date': date.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
